pkg/deploy/deploy.go

```diff
@@ -400,24 +400,44 @@ func SubscriptionExists(cli client.Client, namespace string, name string) (bool,
 	return true, nil
 }
 
-// OperatorExists checks if an Operator with 'operatorPrefix' is installed.
-// Return true if found it, false if not.
-// if we need to check exact version of the operator installed, can append vX.Y.Z later.
-func OperatorExists(cli client.Client, operatorPrefix string) (bool, error) {
+// OperatorExists checks if Operator(list) with 'operatorPrefix' is installed.
+// installed set to true: we expect operator to be installed, installed set to false: we do not want opreator pre-installed
+// If we need to check exact version of the operator installed, can append vX.Y.Z later.
+func OperatorExists(cli client.Client, componentName string, installed bool, operatorPrefixList ...string) (bool, error) {
```

Comment:
A couple of thoughts here

- I would suggest changing the name `operatorPrefixList` -> `operatorPrefixes`
- I assume I can now pass multiple operators to be checked. What does `installed` being set to `false` means in that case? That all are expected to be not present? Maybe we should check `bool` value on the caller side instead as we used to before? This return value is now ignored and it lets us distinguish between not found and error while trying to do so.

Generally speaking, we should keep the amount of params rather small, otherwise, it will make it hard to reason about functions in the future. So besides removing `installed`, if `componentName` is used solely for constructing an error message, maybe we should do it on the caller side instead?

Reply:
`installed` is not accurate, `isExpected` could be better to explain the usage.

for the reason to pass a bool to the function, my main thoughts was:
need an indicator to know 
- if all required operator are presented: this is for the kserve case.
- if none of the operator(s) are presented:  for now codeflare only has one operator which should not be installed, but i am not sure later if we have other component can extend to multiple operators should not be installed